lib/_emerge/BinpkgFetcher.py

```diff
@@ -67,6 +91,16 @@ def _fetcher_exit(self, fetcher):
 			self._fetcher_exit_unlocked(fetcher)
 
 	def _fetcher_exit_unlocked(self, fetcher, unlock_task=None):
+		if not self.pretend and fetcher.returncode != os.EX_OK:
+			if self.retry and len(self.binpkg_format_pending) > 0:
+				# try other formats if current fetcher failed
+				self.binpkg_format = self.binpkg_format_pending.pop()
+				self.pkg_path = pkg.root_config.trees["bintree"].getname(
+					self.pkg.cpv, 
+					binpkg_format=self.binpkg_format) + ".partial"
+				self._start()
+				return
+
```

Comment:
Ideally, the Packages file would tell us which format(s) are available. How about if we add a BINPKG_FORMAT key for each package in the Packages file? We can have a default BINPKG_FORMAT value in the Packages header, and only specify BINPKG_FORMAT for a particular package if it differs from the default.

Reply:
Sounds good, but do we assume that binhost will use the latest version? That will make things easier.

Reply:
Can't you default to xpak when not specified?

Reply:
Updated.